build/ical.js

```diff
@@ -1134,7 +1134,16 @@ ICAL.design = (function() {
     fn: DEFAULT_TYPE_TEXT,
     n: { defaultType: "text", structuredValue: ";", multiValue: "," },
     nickname: DEFAULT_TYPE_TEXT_MULTI,
-    photo: { defaultType: "binary", allowedTypes: ["binary", "uri"] },
+ //   photo: { defaultType: "binary", allowedTypes: ["binary", "uri"] },
+ //   detect binary or http(s) link 
+    photo: { 
+      defaultType: "binary", 
+      allowedTypes: ["binary", "uri"],
+      detectType: function(string) {
+         return ((string.indexOf('https://') === 0) || (string.indexOf('http://') === 0) ? 'uri' : 'binary');
```

Comment:
an uri is not necessarily http(s):// even though common. It could also be a data uri or something else. If type detection is really needed (i.e. lots of non-standard use) please look into a different check.